src/commands/force/source/delete.ts

```diff
@@ -174,7 +176,7 @@ export class Delete extends DeployCommand {
 
   private deleteFilesLocally(): void {
     if (!this.getFlag('checkonly') && getString(this.deployResult, 'response.status') === 'Succeeded') {
-      this.sourceComponents.map((component) => {
+      this.componentSet.toArray().map((component: SourceComponent) => {
```

Comment:
componentSet.toArray is going to have to iterate the whole thing (and it's a generator so it has to call next() on all of them) https://github.com/forcedotcom/source-deploy-retrieve/blob/aa79e2642a7aa35b05cb71c9f4cc0b0c21c21123/src/collections/lazyCollection.ts#L59

Perf idea: do that once and store it somewhere (ex: `this.componentSetAsArray`) to use in the various places.